### PR TITLE
tree-sitter: make tree-sitter cli available.

### PR DESCRIPTION
**Pull request by paperchalice**
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
This commit add necessary build step for tree-sitter-cli and install program tree-sitter.

### CODE REVIEW

File: Formula/tree-sitter.rb [hunk 1]
```diff
@@ -14,9 +14,15 @@ class TreeSitter < Formula
...
   end
 
+  depends_on "rust" => :build
+
   def install
     system "make"
     system "make", "install", "PREFIX=#{prefix}"
+
+    cd "cli" do
+      system "cargo", "install", *std_cargo_args
```
Comment (SMillerDev) [inline Formula/tree-sitter.rb:24]: Please add a test case utilizing this as well

File: Formula/tree-sitter.rb [hunk 2]
```diff
@@ -14,13 +14,59 @@ class TreeSitter < Formula
...
+    end
+  end
+
+  def caveats
+    <<~EOS
+      To automatically create a default config file in ~/.tree-sitter,
```
Comment (SMillerDev) [inline Formula/tree-sitter.rb:30]: This should be in the upstream documentation, not a caveat.

**Comment by paperchalice** [comment]
Some features need calling program in in formula emscripten, should I add emscripten as dependency?

File: Formula/tree-sitter.rb [hunk 3]
```diff
@@ -14,13 +14,52 @@ class TreeSitter < Formula
...
+        name: 'YOUR_LANGUAGE_NAME',
+        rules: {
+          source_file: $ => 'hello'
+        }
+      });
+    EOS
+    system "#{bin}/tree-sitter", "generate"
+
+    # test `tree-sitter parse`
+    (testpath/"test"/"corpus"/"hello.txt").write <<~EOS
```
Comment (SMillerDev) [inline Formula/tree-sitter.rb:45]:
```suggestion
    (testpath/"test/corpus/hello.txt").write <<~EOS
```

**Review by SMillerDev — COMMENTED** [review]
If the cli needs empscripten, add it.

**Review by SMillerDev — APPROVED** [review]
Thanks @paperchalice ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**